lib/webauthn/authenticator_response.rb

```diff
@@ -80,8 +80,12 @@ def valid_origin?(expected_origin)
       expected_origin && (client_data.origin == expected_origin)
     end
 
-    def valid_rp_id?(rp_id)
-      OpenSSL::Digest::SHA256.digest(rp_id) == authenticator_data.rp_id_hash
+    def valid_rp_id?(rp_id, fido_app_id)
+      if fido_app_id_from_extension
+        OpenSSL::Digest::SHA256.digest(fido_app_id.to_s) == authenticator_data.rp_id_hash
```

Comment:
Why is that we check the appid in the extension but don't use it on the comparison?

Reply:
Because it is user input, isn't it?